File: scripts/download_images.py
import os
import requests
from PIL import Image
from io import BytesIO
import logging

# Set up logging
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

# Configuration
IMAGE_DIR = "public/images/menu"
QUALITY = 85
MIN_WIDTH = 1920
MIN_HEIGHT = 1080
IMAGE_FORMAT = "webp"

# Create directory if it doesn't exist
os.makedirs(IMAGE_DIR, exist_ok=True)

def download_and_optimize_image(url, filename):
    """
    Downloads an image from URL and optimizes it according to our requirements
    """
    try:
        # Download image
        response = requests.get(url)
        response.raise_for_status()
        
        # Open image with Pillow
        img = Image.open(BytesIO(response.content))
        
        # Convert to RGB if necessary
        if img.mode in ('RGBA', 'P'):
            img = img.convert('RGB')
        
        # Calculate new dimensions while maintaining aspect ratio
        aspect_ratio = img.width / img.height
        new_width = max(MIN_WIDTH, img.width)
        new_height = int(new_width / aspect_ratio)
        
        if new_height < MIN_HEIGHT:
            new_height = MIN_HEIGHT
            new_width = int(new_height * aspect_ratio)
        
        # Resize image
        img = img.resize((new_width, new_height), Image.Resampling.LANCZOS)
        
        # Save as WebP
        output_path = os.path.join(IMAGE_DIR, f"{filename}.{IMAGE_FORMAT}")
        img.save(output_path, format=IMAGE_FORMAT, quality=QUALITY)
        logger.info(f"Successfully processed {filename}")
        
        return output_path
    
    except Exception as e:
        logger.error(f"Error processing {filename}: {str(e)}")
        return None

# Image URLs to download (to be replaced with actual URLs)
images_to_download = {
    # 분식 (Korean Street Food)
    "cheese-tteokbokki": "https://example.com/tteokbokki.jpg",
    "rabokki": "https://example.com/rabokki.jpg",
    "tempura-set": "https://example.com/tempura.jpg",
    
    # 핫도그 (Hot Dogs)
    "crispy-hotdog": "https://example.com/crispy-hotdog.jpg",
    "cheese-hotdog": "https://example.com/cheese-hotdog.jpg",
    "potato-hotdog": "https://example.com/potato-hotdog.jpg",
    
    # 타코 (Tacos)
    "bulgogi-taco": "https://example.com/bulgogi-taco.jpg",
    "dakgalbi-taco": "https://example.com/dakgalbi-taco.jpg",
    "shrimp-taco": "https://example.com/shrimp-taco.jpg",
    
    # 커피/음료 (Beverages)
    "americano": "https://example.com/americano.jpg",
    "cafe-latte": "https://example.com/latte.jpg",
    "strawberry-smoothie": "https://example.com/smoothie.jpg"
}

def main():
    """
    Main function to process all images
    """
    logger.info("Starting image download and optimization process...")
    
    for filename, url in images_to_download.items():
        output_path = download_and_optimize_image(url, filename)
        if output_path:
            logger.info(f"Saved optimized image to {output_path}")
        else:
            logger.error(f"Failed to process {filename}")
    
    logger.info("Image processing completed")

if __name__ == "__main__":
    main() 

File: scripts/download-images.js
const fs = require('fs');
const path = require('path');
const https = require('https');
const sharp = require('sharp');
const fetch = require('node-fetch');

// Configuration
const IMAGE_DIR = path.join(process.cwd(), 'public', 'images');
const QUALITY = 85;
const MIN_WIDTH = 1920;
const MIN_HEIGHT = 1080;
const IMAGE_FORMAT = 'webp';

// Create directory if it doesn't exist
if (!fs.existsSync(IMAGE_DIR)) {
  fs.mkdirSync(IMAGE_DIR, { recursive: true });
}

// Function to download image
async function downloadImage(url) {
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error(`Failed to download: ${response.status}`);
  }
  return response.buffer();
}

// Function to process image
async function processImage(buffer, filename, options = {}) {
  try {
    const image = sharp(buffer);
    const metadata = await image.metadata();

    // Calculate new dimensions
    const aspectRatio = metadata.width / metadata.height;
    let newWidth = Math.max(options.width || MIN_WIDTH, metadata.width);
    let newHeight = Math.round(newWidth / aspectRatio);

    if (newHeight < (options.height || MIN_HEIGHT)) {
      newHeight = options.height || MIN_HEIGHT;
      newWidth = Math.round(newHeight * aspectRatio);
    }

    // Process and save image
    await image
      .resize(newWidth, newHeight)
      .webp({ quality: QUALITY })
      .toFile(path.join(IMAGE_DIR, `${filename}.${IMAGE_FORMAT}`));

    console.log(`Successfully processed ${filename}`);
  } catch (error) {
    console.error(`Error processing ${filename}:`, error);
  }
}

// Images to download with parameters for better quality
const images = {
  // 메인 페이지 이미지
  'hero-background': {
    url: 'https://images.unsplash.com/photo-1565299585323-38d6b0865b47',
    width: 1920,
    height: 1080
  },
  'about-1': {
    url: 'https://images.unsplash.com/photo-1565299624946-b28f40a0ae38',
    width: 800,
    height: 600
  },
  'about-2': {
    url: 'https://images.unsplash.com/photo-1565299507177-b0ac66763828',
    width: 800,
    height: 600
  },
  'service-1': {
    url: 'https://images.unsplash.com/photo-1565299586001-b5098a3432cf',
    width: 800,
    height: 600
  },
  'service-2': {
    url: 'https://images.unsplash.com/photo-1565299501941-7e5d5f2c0a13',
    width: 800,
    height: 600
  },
  
  // 카테고리 이미지
  'popular-category': {
    url: 'https://images.unsplash.com/photo-1583032015879-e5022cb87c3b',
    width: 800,
    height: 600
  },
  'bunsik-category': {
    url: 'https://images.unsplash.com/photo-1580651315530-69c8e0026377',
    width: 800,
    height: 600
  },
  'hotdog-category': {
    url: 'https://images.unsplash.com/photo-1612392062631-94dd858cba88',
    width: 800,
    height: 600
  },
  'taco-category': {
    url: 'https://images.unsplash.com/photo-1551504734-5ee1c4a1479b',
    width: 800,
    height: 600
  },
  'beverage-category': {
    url: 'https://images.unsplash.com/photo-1514432324607-a09d9b4aefdd',
    width: 800,
    height: 600
  },
  
  // 메뉴 이미지
  'cheese-tteokbokki': {
    url: 'https://images.unsplash.com/photo-1580651315530-69c8e0026377',
    width: 800,
    height: 600
  },
  'rabokki': {
    url: 'https://images.unsplash.com/photo-1580651315530-69c8e0026377',
    width: 800,
    height: 600
  },
  'tempura-set': {
    url: 'https://images.unsplash.com/photo-1476224203421-9ac39bcb3327',
    width: 800,
    height: 600
  },
  'crispy-hotdog': {
    url: 'https://images.unsplash.com/photo-1612392062631-94dd858cba88',
    width: 800,
    height: 600
  },
  'cheese-hotdog': {
    url: 'https://images.unsplash.com/photo-1619740455993-9e612b1af08a',
    width: 800,
    height: 600
  },
  'potato-hotdog': {
    url: 'https://images.unsplash.com/photo-1619740455993-9e612b1af08a',
    width: 800,
    height: 600
  },
  'bulgogi-taco': {
    url: 'https://images.unsplash.com/photo-1551504734-5ee1c4a1479b',
    width: 800,
    height: 600
  },
  'dakgalbi-taco': {
    url: 'https://images.unsplash.com/photo-1565299585323-38d6b0865b47',
    width: 800,
    height: 600
  },
  'shrimp-taco': {
    url: 'https://images.unsplash.com/photo-1611250188496-e966043a0629',
    width: 800,
    height: 600
  }
};

// Main function
async function main() {
  console.log('Starting image download and optimization process...');

  for (const [filename, imageConfig] of Object.entries(images)) {
    try {
      console.log(`Processing ${filename}...`);
      const buffer = await downloadImage(`${imageConfig.url}?w=${imageConfig.width}&h=${imageConfig.height}&fit=crop&q=85`);
      await processImage(buffer, filename, {
        width: imageConfig.width,
        height: imageConfig.height
      });
    } catch (error) {
      console.error(`Failed to process ${filename}:`, error);
    }
  }

  console.log('Image processing completed');
}

main().catch(console.error); 